pkg/plugins/policies/meshtimeout/api/v1alpha1/validator.go

```diff
@@ -18,19 +20,30 @@ func (r *MeshTimeoutResource) validate() error {
 	return verr.OrNil()
 }
 
-func validateTop(targetRef common_api.TargetRef) validators.ValidationError {
-	targetRefErr := mesh.ValidateTargetRef(targetRef, &mesh.ValidateTargetRefOpts{
-		SupportedKinds: []common_api.TargetRefKind{
-			common_api.Mesh,
-			common_api.MeshSubset,
-			common_api.MeshGateway,
-			common_api.MeshService,
-			common_api.MeshServiceSubset,
-			common_api.MeshHTTPRoute,
-		},
-		GatewayListenerTagsAllowed: true,
-	})
-	return targetRefErr
+func (r *MeshTimeoutResource) validateTop(targetRef common_api.TargetRef) validators.ValidationError {
+	switch core_model.PolicyRole(r) {
+	case mesh_proto.SystemPolicyRole:
+		return mesh.ValidateTargetRef(targetRef, &mesh.ValidateTargetRefOpts{
+			SupportedKinds: []common_api.TargetRefKind{
+				common_api.Mesh,
+				common_api.MeshSubset,
+				common_api.MeshGateway,
+				common_api.MeshService,
+				common_api.MeshServiceSubset,
+				common_api.MeshHTTPRoute,
+			},
+			GatewayListenerTagsAllowed: true,
+		})
+	default:
+		return mesh.ValidateTargetRef(targetRef, &mesh.ValidateTargetRefOpts{
+			SupportedKinds: []common_api.TargetRefKind{
+				common_api.Mesh,
```

Comment:
shouldn't we allow targeting `MeshHttpRoute`? Or will it be changed in the future with full support for policies on namespace?

Reply:
TBH I don't want to allow MeshHTTPRoute in top-level targetRef of namespace-scoped policies to deprecated in the near future with https://github.com/kumahq/kuma/issues/10247.